File: 2015/day1/part2.swift
func part2(_ input: String) -> Int {
	var currentFloor = 0
	var ret: Int = 0
	for (i, ch) in input.enumerated() {
		switch ch {
			case "(": currentFloor += 1
			case ")": currentFloor -= 1
			default: break
		}
		if currentFloor == -1 {
			ret = i + 1
			break
		}
	}
	return ret
}

File: 2015/day1/part2.py
def part2(floors):
    curr_floor = 0
    ret = 0
    for i, ch in enumerate(floors):
        if ch == '(':
            curr_floor += 1
        elif ch == ')':
            curr_floor -= 1
        if curr_floor == -1:
            ret = i + 1
            break
    return ret

print(part2(')')) # 1
print(part2('()())')) # 5